x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.gen.ts

```diff
@@ -205,3 +321,78 @@ export const RelatedIntegration = z.object({
 
 export type RelatedIntegrationArray = z.infer<typeof RelatedIntegrationArray>;
 export const RelatedIntegrationArray = z.array(RelatedIntegration);
+
+export type InvestigationFields = z.infer<typeof InvestigationFields>;
+export const InvestigationFields = z.object({
+  field_names: z.array(z.string()).min(1),
+});
+
+/**
+ * Defines the interval on which a rule's actions are executed.
+ */
+export type RuleActionThrottle = z.infer<typeof RuleActionThrottle>;
+export const RuleActionThrottle = z.union([
+  z.enum(['no_actions', 'rule']),
+  z.string().regex(/^[1-9]\d*[smhd]$/),
+]);
+
+/**
+ * The action frequency defines when the action runs (for example, only on rule execution or at specific time intervals).
+ */
+export type RuleActionFrequency = z.infer<typeof RuleActionFrequency>;
+export const RuleActionFrequency = z.object({
+  summary: z.boolean(),
+  notifyWhen: z.enum(['onActionGroupChange', 'onActiveAlert', 'onThrottleInterval']),
+  throttle: RuleActionThrottle.nullable(),
+});
+
+export type RuleAction = z.infer<typeof RuleAction>;
+export const RuleAction = z.object({
```

Comment:
We have a `RuleActionArray` type which is not defined here in the new common attributes file, but is still defined in `packages/kbn-securitysolution-io-ts-alerting-types/src/actions/index.ts`, and a lot of files depend on it.

I'd like to get an understanding of how much cleanup in general will be left to do after we merge this PR and migrate the rest of the Rule Management API endpoints. Because it feels like a big chunk of work.

Let's also add comments to `packages/kbn-securitysolution-io-ts-alerting-types/src/actions/index.ts` and other files in `kbn-securitysolution-*` packages where we have any legacy rule schemas that these files are deprecated, a migration is in progress, and no new schemas should be added until the migration is done.

Reply:
`@deprecated` attribute with a comment will work even better.